src/transformers/tokenization_utils_base.py

```diff
@@ -1920,6 +1921,14 @@ def convert_added_tokens(obj: Union[AddedToken, Any], add_type_field=True):
 
         # add_type_field=True to allow dicts in the kwargs / differentiate from AddedToken serialization
         tokenizer_config = convert_added_tokens(tokenizer_config, add_type_field=True)
+
+        # Add tokenizer class to the tokenizer config to be able to reload it with from_pretrained
+        tokenizer_class = self.__class__.__name__
+        # Remove the Fast at the end unless we have a special `PreTrainedTokenizerFast`
+        if tokenizer_class.endswith("Fast") and tokenizer_class != "PreTrainedTokenizerFast":
+            tokenizer_class = tokenizer_class[:-4]
+        tokenizer_config["tokenizer_class"] = tokenizer_class
```

Comment:
This is the main bit that allows to do `AutoTokenizer.from_pretrained` after a `save_pretrained`, since it stores the tokenizer class in the tokenizer config.

Reply:
Cool, this is very nice!